🐛 BUG: typegen command errors when the entry point does not exist

### Which Cloudflare product(s) does this pertain to?

Wrangler

### What version(s) of the tool(s) are you using?

3.90.0 [Wrangler], latest version of C3

### What version of Node are you using?

_No response_

### What operating system and version are you using?

MacOS Version 15.1.1 (24B91)

### Describe the Bug

### Observed behavior

With `@opennextjs/cloudflare`, the `npm run cf-typegen` command throws the following error:

```
✘ [ERROR] The entry-point file at ".worker-next/index.mjs" was not found.
```

This is because the entry point is not generated until you run the build command

### Expected behavior

The correct types should get added to the `env.d.ts` file.

It looks like the entry point is parsed to determine service worker vs module worker format.

We should default ot module when the ep is not found

### Steps to reproduce
Please provide the following:
- Create a new Next.js project using C3 with the `--experimental` flag: `npx create cloudflare@latest next-app -- --framework=next --experimental`
- Add a binding in `wrangler.toml` or add a variable in `.dev.vars`.
- Run the command `npx run cf-typegen`

This will throw an error.


### Please provide a link to a minimal reproduction

_No response_

### Please provide any relevant error logs

_No response_

--- a/.changeset/silly-news-compete.md
+++ b/.changeset/silly-news-compete.md
@@ -1,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+fix: allow running `wrangler types` when expected entrypoint doesn't exist

--- a/packages/wrangler/src/__tests__/type-generation.test.ts
+++ b/packages/wrangler/src/__tests__/type-generation.test.ts
@@ -540,6 +540,31 @@ describe("generateTypes()", () => {
 	`);
 	});
 
+	it("should not error if expected entrypoint is not found and assume module worker", async () => {
+		fs.writeFileSync(
+			"./wrangler.toml",
+			TOML.stringify({
+				main: "index.ts",
+				vars: bindingsConfigMock.vars,
+			} as unknown as TOML.JsonMap),
+			"utf-8"
+		);
+		// note index.ts does not exist
+
+		await runWrangler("types");
+		expect(std.out).toMatchInlineSnapshot(`
+		"Generating project types...
+
+		interface Env {
+			SOMETHING: \\"asdasdfasdf\\";
+			ANOTHER: \\"thing\\";
+			\\"some-other-var\\": \\"some-other-value\\";
+			OBJECT_VAR: {\\"enterprise\\":\\"1701-D\\",\\"activeDuty\\":true,\\"captian\\":\\"Picard\\"};
+		}
+		"
+	`);
+	});
+
 	it("should include secret keys from .dev.vars", async () => {
 		fs.writeFileSync(
 			"./wrangler.toml",

--- a/packages/wrangler/src/__tests__/type-generation.test.ts
+++ b/packages/wrangler/src/__tests__/type-generation.test.ts
@@ -549,7 +549,7 @@ describe("generateTypes()", () => {
 			} as unknown as TOML.JsonMap),
 			"utf-8"
 		);
-		// note index.ts does not exist
+		expect(fs.existsSync("index.ts")).toEqual(false);
 
 		await runWrangler("types");
 		expect(std.out).toMatchInlineSnapshot(`

--- a/packages/wrangler/src/type-generation/index.ts
+++ b/packages/wrangler/src/type-generation/index.ts
@@ -12,6 +12,7 @@ import { printWranglerBanner } from "../update-check";
 import { generateRuntimeTypes } from "./runtime";
 import { logRuntimeTypesMessage } from "./runtime/log-runtime-types-message";
 import type { Config } from "../config";
+import type { Entry } from "../deployment-bundle/entry";
 import type { CfScriptFormat } from "../deployment-bundle/worker";
 import type {
 	CommonYargsArgv,
@@ -232,10 +233,17 @@ async function generateTypes(
 	const configContainsEntrypoint =
 		config.main !== undefined || !!config.site?.["entry-point"];
 
-	const entrypoint = configContainsEntrypoint
-		? await getEntry({}, config, "types")
-		: undefined;
-
+	let entrypoint: Entry | undefined;
+	if (configContainsEntrypoint) {
+		// this will throw if an entrypoint is expected, but doesn't exist
+		// e.g. before building. however someone might still want to generate types
+		// so we default to module worker
+		try {
+			entrypoint = await getEntry({}, config, "types");
+		} catch {
+			entrypoint = undefined;
+		}
+	}
 	const entrypointFormat = entrypoint?.format ?? "modules";
 	const fullOutputPath = resolve(outputPath);
 

